engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -1264,12 +1264,12 @@ public Operator loadOperator(PTOperator oper)
    */
   private Checkpoint getActivationCheckpoint(PTOperator oper)
   {
-    if (oper.recoveryCheckpoint == null && oper.checkpoints.isEmpty()) {
+    if ((oper.recoveryCheckpoint == null || oper.recoveryCheckpoint == Checkpoint.INITIAL_CHECKPOINT) && oper.checkpoints.isEmpty()) {
```

Comment:
I believe this check should be removed and instead null passed to addPTOperator. Please verify.

```
+++ b/engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java
@@ -781,7 +781,7 @@ public class PhysicalPlan implements Serializable
     // create operator instance per partition
     Map<Integer, Partition<Operator>> operatorIdToPartition = Maps.newHashMapWithExpectedSize(partitions.size());
     for (Partition<Operator> partition : partitions) {
-      PTOperator p = addPTOperator(m, partition, Checkpoint.INITIAL_CHECKPOINT);
+      PTOperator p = addPTOperator(m, partition, null);
       operatorIdToPartition.put(p.getId(), partition);
     }

@@ -1264,7 +1264,7 @@ public class PhysicalPlan implements Serializable
    */
   private Checkpoint getActivationCheckpoint(PTOperator oper)
   {
-    if ((oper.recoveryCheckpoint == null || oper.recoveryCheckpoint == Checkpoint.INITIAL_CHECKPOINT) && oper.checkpoints.isEmpty()) {
+    if (oper.recoveryCheckpoint == null && oper.checkpoints.isEmpty()) {
       Checkpoint activationCheckpoint = Checkpoint.INITIAL_CHECKPOINT;
       for (PTInput input : oper.inputs) {
         PTOperator sourceOper = input.source.source;
```